extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/MSQTasks.java

```diff
@@ -114,7 +114,13 @@ public static long primaryTimestampFromObjectForInsert(final Object timestamp)
       // be a long at execution time. So a nice user-friendly message isn't needed here: it would only happen
       // if the SQL layer is bypassed. Nice, friendly users wouldn't do that :)
       final UnknownFault fault =
-          UnknownFault.forMessage(StringUtils.format("Incorrect type for [%s]", ColumnHolder.TIME_COLUMN_NAME));
+          UnknownFault.forMessage(
+              StringUtils.format(
+                  "Incorrect type for column [%s]. Expected [Long] but got [%s]. Please ensure the value is casted to [Long].",
```

Comment:
Extrapolations are incorrectly used - 

1. Anything known is not an extrapolation. `[Long]` should not be in extrapolation
2. The sentence should make sense if extrapolations are removed. Second "[%s]" would break the flow of the sentence. Think of it like the `,,` in the English grammar. `

```suggestion
                  "Incorrect type for column [%s]. Expected LONG but got type [%s]. Please ensure that the value is cast to LONG.",
```

Reply:
https://github.com/apache/druid/blob/master/dev/style-conventions.md#message-formatting-for-logs-and-exceptions

Reply:
@LakshSingla I see, thanks for pointing this out!

I was trying to keep the error message format consistent for the user (for example: `Expected [type1] but got [type2]` irrespective of whether these are extrapolated or not, since the final message for the end user is what matters IMO).

Have made the suggested change though, will keep this in mind for future as well!